### PR TITLE
MCP3002 support

### PR DESCRIPTION
**Pull request by 2bndy5**
- added MCP3002 support via new module and corresponding examples to address #2 
### docs improvements
Since I had to test differential reads pretty thoroughly, I ended up 
- making the docs a bit more concise concerning the differential channel mappings and the driver class' channel aliases
- added return value ranges to applicable attributes
- added some links to datasheets for wiring/pinout (yea I screwed it up on the first try)
- `AnalogIn`'s API contains a warning about the ADCs' neglect of negative numbers for differential read results and hints behavior for voltages outside reference voltage bounds. Although, I feel the placement of this warning might serve better at the top of the API references, but its fine for now.
### code improvements
- MCP3002's SPI calls transact over 2 bytes while MCP3004/MCP3008 still transact over 3 bytes
- removed redundantly setting each byte in the MCP3xxx's internal output buffer and removed the local variable `command`
- removed the constants used for the differential read flag & the start bit as the start bit is now set upon instantiation while differential read flag is still set during every `read()` call.
- modified the handling of the received data in the input buffer to use 1 less computation and removed local variable `result`
- removed unused object instance attributes `negative_pin` and `channel` from `AnalogIn` class as well as the `__getitem__()` related to the `channel` attribute.

All these changes were tested on a MCP3002 and MCP3008 for verification with consistent results.

### CODE REVIEW

File: adafruit_mcp3xxx/analog_in.py [hunk 1]
```diff
@@ -26,40 +26,44 @@
...
+        """Returns the value of an ADC pin as an integer. Due to 10-bit accuracy of the chip, the
+        returned values range [0, 65472]."""
         return self._mcp.read(self._pin_setting, is_differential=self.is_differential) << 6
 
     @property
     def voltage(self):
-        """Returns the voltage from the ADC pin as a floating point value."""
+        """Returns the voltage from the ADC pin as a floating point value. Due to the 10-bit
+        accuracy of the chip, returned values range from 0 to (``reference_voltage`` *
+        65472 / 65535)"""
```
Comment (brentru) [inline adafruit_mcp3xxx/analog_in.py]: Wouldn't returned values be:

> returned values range from 0 to (``adc_pin_value`` *
        ``reference_voltage`` / 65535)"""

File: adafruit_mcp3xxx/analog_in.py [hunk 2]
```diff
@@ -26,40 +26,44 @@
...
-            self._pin_setting = self._pins.get((self._pin_setting, self._negative_pin),
-                                               "Difference pin not found.")
-
-    def __getitem__(self, key):
-        return self._channels[self._pins[key]]
+        self.is_differential = negative_pin is not None
+        if self.is_differential:
+            self._pin_setting = self._mcp.DIFF_PINS.get((positive_pin, negative_pin), None)
+            if self._pin_setting is None:
+                raise ValueError("Differential pin mapping not defined. Please read the docs.")
```
[the remark this answers is not in the document]
Comment (2bndy5) [reply adafruit_mcp3xxx/analog_in.py]: Do you mean replace the "please read the docs" with "please read https://circuitpython.readthedocs.io/projects/mcp3xxx/en/latest/api.html#"

File: adafruit_mcp3xxx/analog_in.py (hunk 1 above)
Comment (2bndy5) [reply adafruit_mcp3xxx/analog_in.py]: I'm really trying to state just the max and min values, so using the `adc_pin_value` would mean `1023` which gets translated into `65472`.

Comment (brentru) [reply adafruit_mcp3xxx/analog_in.py]: wouldnt the `value` of the ADC pin be the pin reading from 0 to 65,472?

File: adafruit_mcp3xxx/analog_in.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (brentru) [reply adafruit_mcp3xxx/analog_in.py]: Maybe something like `Please read the documentation for valid differential channel mappings`?

File: adafruit_mcp3xxx/analog_in.py (hunk 1 above)
Comment (2bndy5) [reply adafruit_mcp3xxx/analog_in.py]: Yes and that's what I put in the docs for the `value` attribute, but this line (`voltage` attribute) depends on the `reference_voltage` passed to `MCP3xxx` c'tor

File: adafruit_mcp3xxx/analog_in.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (2bndy5) [reply adafruit_mcp3xxx/analog_in.py]: Sure, no problem

File: adafruit_mcp3xxx/analog_in.py (hunk 1 above)
Comment (brentru) [reply adafruit_mcp3xxx/analog_in.py]: Ah, ok, I misread. Looks good then.

**Comment by brentru** [comment]
@2bndy5  Travis is passing, merging in and releasing! Thanks for the contribution, this should close  https://github.com/adafruit/Adafruit_CircuitPython_MCP3xxx/issues/2

**Pull request merged**